Proposal: Add ephemeral flag to wakuMessage (#532)
vacp2p/rfc: content/docs/rfcs/13/README.md

The high uptime requirement makes sure that no message is missed out hence a complete and intact view of the message history is delivered to the querying nodes.
Nevertheless, in case that storage provider nodes cannot afford high availability, the querying nodes may retrieve the historical messages from multiple sources to achieve a full and intact view of the past.

> The concept of "ephemeral" messages exists while using `13/WAKU2-STORE` as well.

Reviewer: ```suggestion
The concept of "ephemeral" messages introduced in `[14/WAKU2-MESSAGE](/spec/14)` affects `13/WAKU2-STORE` as well.
```

Reviewer: I'd adjust the introductory sentence similar to this, because in the current version, it immediatly clear what "ephemeral messages" refers to in this context.

Author: Addressed in [f143aa3](https://github.com/vacp2p/rfc/pull/532/commits/f143aa315dcc5745beedf149afbed8b4c15c0a8f)

Nodes running [13/WAKU2-STORE](/spec/13) SHOULD support "ephemeral" messages as specified in [14/WAKU2-MESSAGE](/spec/14).

> Nodes running [13/WAKU2-STORE](/spec/13) SHOULD NOT not store messages with the `ephemeral` flag set to `true`.

Reviewer: ```suggestion
Nodes running [13/WAKU2-STORE](/spec/13) SHOULD NOT store messages with the `ephemeral` flag set to `true`.
```

Author: Addressed in [9a4b4fd](https://github.com/vacp2p/rfc/pull/532/commits/9a4b4fd7072d2e2ed372f4ac13c189517d771932)

The high uptime requirement makes sure that no message is missed out hence a complete and intact view of the message history is delivered to the querying nodes.
Nevertheless, in case that storage provider nodes cannot afford high availability, the querying nodes may retrieve the historical messages from multiple sources to achieve a full and intact view of the past.

> The concept of "ephemeral" messages exists while using `13/WAKU2-STORE` as well.

Reviewer: I'd explicitly make the connection to the message spec and introduce "ephemeral" messages suppport as a "SHOULD".
Maybe like so:

> Nodes running [13/WAKU2-STORE](https://rfc.vac.dev/spec/13/) SHOULD support "ephemeral" messages as specified in [14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/).

And then explain what it means to support "ephemeral" messages.

In the process of #528, we should also add this to the protocol specification section.

Author: Should I be reformatting the rfc's to comply with the [RFC Template](https://github.com/vacp2p/rfc/blob/master/content/docs/rfcs/template/README.md) as described in #528 , or is that for another PR?

Author: Partially addressed in [0ca7525](https://github.com/vacp2p/rfc/pull/532/commits/0ca7525cfa855c8f2c88cf95c96032956a379602)

Reviewer: > Should I be reformatting the rfc's to comply with the RFC Template as described in #528 , or is that for another PR?

This should be done in another PR, as it affects other parts, outside of the ephemeral flag field, too.

> Any `WakuMessage` that has the `ttl` field set to `0` MAY be ignored by the store service node, thereby reducing the amount of one-time use messages stored.

Reviewer: There is no `ttl` field in the protobuf. Where is the ttl coming from?

Author: I have reverted the changes the `ttl` proposal brings. Addressed in [bb4989d](https://github.com/vacp2p/rfc/pull/532/commits/bb4989d6ec3a5dda2d7f2ef3d28176c9f52193ef)

The high uptime requirement makes sure that no message is missed out hence a complete and intact view of the message history is delivered to the querying nodes.
Nevertheless, in case that storage provider nodes cannot afford high availability, the querying nodes may retrieve the historical messages from multiple sources to achieve a full and intact view of the past.

> The concept of "ephemeral" messages exists while using `13/WAKU2-STORE` as well. Any `WakuMessage` that has the ephemeral flag set to true will not be added to the store, thereby reducing the amount of one-time use messages on disk. While the node SHOULD not store messages with the ephemeral flag set to true, they MAY do so for archival.

Reviewer 2: We try to use [semantic line breaks](https://sembr.org/#:~:text=A%20semantic%20line%20break%20MUST%20occur%20after%20a%20sentence%2C%20as,or%20em%20dash%20(%E2%80%94)) in RFCs.

If this sentence is about the benefit this flag provides, perhaps instead of saying "will not be added to the store" we can say something like "MAY be ignored by the store service node". I will also remove "on disk" (in-memory stores exist too). The idea is to indicate to `store` nodes how to interpret the protocol, not dictate implementation.

Reviewer 2: ```suggestion
The concept of "ephemeral" messages exists while using `13/WAKU2-STORE` as well. Any `WakuMessage` that has the ephemeral flag set to true will not be added to the store, thereby reducing the amount of one-time use messages on disk. Nodes SHOULD NOT store messages with the `ephemeral` flag set to `true`.
```

No need (at spec level) to indicate why nodes may still choose to do so. The fact that it's optional (but not recommended) is implied by "SHOULD NOT".

Author: Addressed in [eeaa58b](https://github.com/vacp2p/rfc/pull/532/commits/eeaa58b797d98109049b8d616bcce89e0e354468)
